docs/How-to-implement-a-new-ci-test.md

```diff
@@ -0,0 +1,146 @@
+
+
+Now following are the necessary steps for making a new distribution acessible in GPUCSL. We recommended to follow the given order of steps.
+
+
+## Implement your CUDA kernel
+
+This is the part we can help you with the least. You have to design efficient CUDA kernel code. While doing so you can use 
+any of the already existing helper functions and you can reuse code from our distributions however you like.
+
+You have to place your cuda files in the gpucsl/pc/cuda or gpucsl/pc/cuda/helpers directories! (otherwise you have to modify
+the compiler_options variable in the get_module function of gpucsl/pc/kernel_management.py like it is done for cuda_dir or cuda_helpers_dir)
+
+Note: If you want you can use templates. As the CUDA code is compiled during the runtime of GPUCSL (!) there are already some 
+runtime informations, like the number of variables, accessible. For more information see the next section.
+
+
+## Implement your own Kernel Management
+
+The Kernel Management encapsulates the details of kernel access, for example the grid and block mapping of the kernel is defined here.
+Otherwise encapsulates details of the underlying cupy libraray (which cuda file to read, it compiles the code on creation of the class, function name resulution to level). 
+
+To write your own you want inherit from the abstract base class Kernel. You then have to implement the following methods:
+
+- __init__: With the __init__ method you take the paramters you need for everything involved into your kernel management. So if you need something to calculate your grid and block mapping you need to take this parameters here. At a minimum you have to take the paramters is_debug and should_log.
+  A template for an __init__ method is:
+  ```
+  def __init__(
+        self,
+        your_choosen_parameters,
+        is_debug: bool = False,
+        should_log: bool = False,
+    ):
+        super().__init__(is_debug=is_debug, should_log=should_log)
+        self.your_choosen_parameters = your_choosen_parameters
+
+        # calculate all the kernel function signature names you later on want to be able to call (something needed by cupy)
+        # has to be an array, but if you only need one signature you can change this to be an array with one entry
+        kernel_function_signatures = [
+            self.kernel_function_signature_for_level(level)
+            for level in range(0, max_level + 1)
+        ]
+
+        # pass the kernel function signatures and compile the cuda code
+        self.define_module("gaussian_ci.cu", kernel_function_signatures)
+  ```    
+- kernel_function_name_for_level: Return the name of the kernel for the given level. Can return different names for different levels (as you maybe have written 
+    optimizations for a specific level in form of an extra CUDA function, like we did for DiscreteKernel. Otherwise you can also return just a static name like the CompactKernel)
```

Comment:
```suggestion
    optimizations for a specific level in form of an extra CUDA function, as we did for DiscreteKernel. Otherwise, you can also return just a static name like the CompactKernel)
```